File: csharp/robot-name/RobotName.cs
using System;

public class Robot
{
   private string name;
   
   public Robot()
   {
       Reset();
   }
    public string Name
    {
        get => name;
    }

    public void Reset()
    {
        string letters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        string numbers = "1234567890";
        Random rand = new Random();

        name = GetRandomCharsFromString(letters, 2, rand);
        name += GetRandomCharsFromString(numbers, 3, rand);
    }

    private string GetRandomCharsFromString(string s, int n, Random r)
    {
        string output = "";

        for(int i = 0; i <= n; i++)
        {
            output += s[r.Next(s.Length)];
        }
        return output;
    }
}

File: fsharp/robot-name/RobotName.fs
module RobotName

open System

type Robot = { name: string }

let private getRandomString (ch: char list) (r: Random) (n: int): string = 
    [1..n]
    |> List.map (fun _ -> ch.Item(r.Next(ch.Length)))
    |> List.toArray
    |> String

let private generateName(): string = 
    let letters = ['A'..'Z']
    let numbers = ['0'..'9']
    let rand = Random()
    getRandomString letters rand 2 + 
    getRandomString numbers rand 3

let mkRobot(): Robot = { name = generateName() }

let name (robot: Robot): string = robot.name

let reset (robot: Robot): Robot = mkRobot()